blockmanager.go

```diff
@@ -2006,13 +2148,367 @@ out:
 			}
 
 		case <-b.quit:
-			break out
+			break unCheckPtLoop
 		}
 	}
 
 	log.Trace("Block handler done")
 }
 
+// processBlKHeaderInCheckPtRegionInOrder handles and writes the block headers received from querying the
+// workmanager while fetching headers within the block header checkpoint region. This process is carried out
+// in order.
+func (b *blockManager) processBlKHeaderInCheckPtRegionInOrder() {
```

Comment:
this name reflects a lack of clarity of purpose here. Judging from the rest of the context it seems like this is one of the main coroutines that the blockManager is responsible for running. I'd recommend carving out whatever the responsibility of this coroutine is and giving it its own structure rather than mixing it in with the rest of the block manager.

Reply:
So create a different struct?

Reply:
Maybe. First we need to figure out what this method is doing, what the minimal set of arguments would be and see if we can carve it off of the blockManager. It seems like the appropriate thing here would be to make this function apply to some sort of "batch" structure.

Reply:
I think the comment on top of the function definition would help in understanding what it is about. IMO, I do not think this needs a different struct, if it does then, the `cfhandler` function and the functions/functionality involved in writing the `cfheaders` to the store in order should be in a different struct as well as this is block header version of it. I also think the naming of the function would also help understand what the function is about, it is literally, "processing block header within a check point in order".